SerialPrograms/Source/PokemonSV/Inference/Overworld/PokemonSV_DirectionDetector.cpp

```diff
@@ -0,0 +1,169 @@
+/*  Direction Detector
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/GlobalSettingsPanel.h"
+#include "Common/Cpp/Containers/FixedLimitVector.tpp"
+#include "CommonFramework/ImageMatch/WaterfillTemplateMatcher.h"
+#include "CommonFramework/ImageTypes/ImageViewRGB32.h"
+#include "CommonFramework/ImageTools/WaterfillUtilities.h"
+#include "CommonFramework/ImageTools/BinaryImage_FilterRgb32.h"
+#include "CommonFramework/ImageTools/ImageManip.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "CommonFramework/Logging/Logger.h"
+#include "CommonFramework/Tools/DebugDumper.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "Kernels/Waterfill/Kernels_Waterfill_Types.h"
+#include "PokemonSV_DirectionDetector.h"
+#include <cmath>
+#include <iostream>
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+//using std::cout;
+//using std::endl;
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+
+class DirectionMatcher : public ImageMatch::WaterfillTemplateMatcher{
+public:
+    
+    DirectionMatcher() : WaterfillTemplateMatcher(
+        "PokemonSV/Map/DirectionIcon-Template.png", Color(0, 100, 30), Color(50, 230, 85), 50
+    ){
+        m_aspect_ratio_lower = 0.8;
+        m_aspect_ratio_upper = 1.2;
+        m_area_ratio_lower = 0.8;
+        m_area_ratio_upper = 1.2;
+
+    }
+
+    static const ImageMatch::WaterfillTemplateMatcher& instance(){
+        static DirectionMatcher matcher;
+        return matcher;
+    }
+};
+
+
+DirectionDetector::~DirectionDetector() = default;
+
+DirectionDetector::DirectionDetector(Color color, const ImageFloatBox& box)
+    : m_color(color)
+    , m_minimap_box(box)
+{}
+
+
+std::pair<double, double> DirectionDetector::locate_north(const ImageViewRGB32& screen) const{
+    const std::vector<std::pair<uint32_t, uint32_t>> filters = {
+        {combine_rgb(0, 100, 0), combine_rgb(50, 230, 80)},
+
+    };
+
+    const double min_object_size = 50.0;
+    const double rmsd_threshold = 60.0;
+
+    const double screen_rel_size = (screen.height() / 1080.0);
+    const size_t min_size = size_t(screen_rel_size * screen_rel_size * min_object_size);
+
+    std::pair<double, double> north_location(0, 0);
+    std::pair<double, double> zero_coord(0.9065, 0.8335);
```

Comment:
Is it a good idea to hard-code the coordinate of the radar like this? Since some capture cards may shift by a few pixels.